src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
+const ERROR_MESSAGE = 'Invalid values entered';
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+
+type Color = 'red' | 'blue' | 'green';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+function roundDown(number: number, decimals: number = 2): number {
+  const factor = 10 ** decimals;
 
+  return Math.floor(number * factor) / factor;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(ERROR_MESSAGE);
+    }
+
+    const longestSide = Math.max(a, b, c);
+    const sumOfSides = a + b + c;
 
+    if (sumOfSides - longestSide <= longestSide) {
+      throw new Error(ERROR_MESSAGE);
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+
+    return roundDown(
+      Math.sqrt(
+        semiPerimeter
+          * (semiPerimeter - this.a)
+          * (semiPerimeter - this.b)
+          * (semiPerimeter - this.c),
+      ),
+    );
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error(ERROR_MESSAGE);
+    }
+  }
 
+  getArea(): number {
+    return roundDown(Math.PI * (this.radius ** 2));
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape = Shape.Rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error(ERROR_MESSAGE);
+    }
+  }
+
+  getArea(): number {
+    return roundDown(this.height * this.width);
+  }
+}
 
+export function getInfo(figure: Figure): string {
+  return `A ${figure.color} ${figure.shape} - ${figure.getArea()}`;
```

Comment:
We can use destructuring and create variable for `${figure.getArea()}`. It will be easier to read